### PR TITLE
Use dag helper in whiteflag

### PR DESCRIPTION
**Pull request by muXxer**
This refactoring was done for better readability of whiteflag.

### CODE REVIEW

File: pkg/dag/helper.go [hunk 1]
```diff
@@ -139,8 +140,24 @@ func processStackApprovees(stack *list.List, processed map[string]struct{}, chec
...
+		// load up bundle to retrieve trunk and branch of the head tx
+		cachedBundle := tangle.GetCachedBundleOrNil(currentTxHash)
+		if cachedBundle == nil {
+			return fmt.Errorf("%w: bundle %s of candidate tx %s doesn't exist", tangle.ErrBundleNotFound, cachedTx.GetTransaction().Tx.Bundle, currentTxHash.Trytes())
+		}
+		defer cachedBundle.Release(true)
+
+		cachedBundleHeadTx := cachedBundle.GetBundle().GetHead()
+		defer cachedBundleHeadTx.Release(true)
+		trunkHash = cachedBundleHeadTx.GetTransaction().GetTrunkHash()
```
Comment (alexsporn) [inline pkg/dag/helper.go]: cachedBundle.GetTrunk(true)
cachedBundle.GetBranch(true)

File: pkg/whiteflag/confirmation.go [hunk 2]
```diff
@@ -28,21 +28,25 @@ type ConfirmedMilestoneStats struct {
...
 
 	ts := time.Now()
-	confirmation, err := ComputeConfirmation(tangle.GetMilestoneMerkleHashFunc(), cachedMsBundle.Retain(), milestoneIndex)
+
+	confirmation, err := ComputeConfirmation(tangle.GetMilestoneMerkleHashFunc(), msBundle.GetTailHash())
 	if err != nil {
 		// According to the RFC we should panic if we encounter any invalid bundles during confirmation
 		return nil, fmt.Errorf("confirmMilestone: whiteflag.ComputeConfirmation failed with Error: %v", err)
 	}
+	confirmation.MilestoneIndex = milestoneIndex
```
Comment (alexsporn) [inline pkg/whiteflag/confirmation.go]: Don't like that we set these values in an object that was created in ComputeConfirmation

Comment (muXxer) [reply pkg/whiteflag/confirmation.go]: But the COO doesn't know about its final tail hash before the bundle gets issued... So if we want to reuse the code... what would be the best practice?

Comment (alexsporn) [reply pkg/whiteflag/confirmation.go]: This method returns an object not containing it, and the confirmation.go creates an object with this + index + hash?

**Review by alexsporn — APPROVED** [review]

**Pull request merged**